File: tests/samples/java/Car.java
package samples.java;

public class Car {
    private String brand;
    private String color;

    public Car(String brand, String color) {
        this.brand = brand;
        this.color = color;
    }

    public String getBrand() {
        return brand;
    }

    public String getColor() {
        return color;
    }

    public void startEngine() {
        System.out.println("Engine started");
    }

    public void stopEngine() {
        System.out.println("Engine stopped");
    }
}

File: tests/samples/rust/car.rs
struct Car {
    brand: String,
    color: String,
    year: u32,
}

struct Vehicle {
    make: String,
    model: String,
}

impl Car {
    fn new(brand: String, color: String, year: u32) -> Self {
        Car { brand, color, year }
    }
    
    fn start_engine(&self) {
        println!("The {} {} engine has started.", self.color, self.brand);
    }
    
    fn stop_engine(&self) {
        println!("The {} engine has stopped.", self.brand);
    }
}

fn main() {
    let car = Car::new("Toyota".to_string(), "Red".to_string(), 2020);
    car.start_engine();
    car.stop_engine();
}